core/interop/src/loaders/functions.rs

```diff
@@ -0,0 +1,126 @@
+//! This module contains types that help create custom module loaders from functions.
+use boa_engine::module::{resolve_module_specifier, ModuleLoader, Referrer};
+use boa_engine::{Context, JsError, JsNativeError, JsResult, JsString, Module, Source};
+use std::io::Cursor;
+
+/// Create a [`ModuleLoader`] from a function that takes a referrer and a path,
+/// and returns a [Module] if it exists, or an error.
+///
+/// This function cannot be `async` and must be blocking. An `async` version of
+/// this code will likely exist as a separate function in the future.
+///
+/// `F` cannot be a mutable closure as it could recursively call itself.
+#[derive(Copy, Clone)]
+pub struct FnModuleLoader<F>(F, &'static str)
```

Comment:
I think it would be clearer if we used named field, especially for structs with more than one field.